### ISSUE TITLE
Landscape Canvas: The Editor crashes when adding a node to the graph, after creating a prefab out of an Entity with the Landscape Canvas component

### ISSUE THREAD

**Issue by LB-JakubSkorupka** [o3de/o3de#4169]
**Describe the bug**
The Editor crashes when adding a node to the **Landscape Canvas** graph after creating a prefab out of Entity with **Landscape Canvas** component.
Please note that before crash occurs the "Entity Creation Error" window is displayed:
"Cannot add entity because the owning instance of parent entity with id 'XXX' could not be found."

Pressing the X in the right top corner, or "Ok" to close the Error results in Editor crashing.

**To Reproduce**
1. Create a new **Landscape Canvas** graph.
2. Create a prefab out of an Entity created with the **Landscape Canvas** component.
3. Add a node to the **Landscape Canvas** graph.
4. Press X or "Ok" to close the "Entity Creation Error" window.

**Expected behavior**
The Editor remains stable.

**Call Stack Trace:**
```
25) 00007FFBC2B2C247 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFBC5395270 (ntdll) : memset
23) 00007FFBC537C776 (ntdll) : _C_specific_handler
22) 00007FFBC539217F (ntdll) : _chkstk
21) 00007FFBC5341454 (ntdll) : RtlRaiseException
20) 00007FFBC5390CAE (ntdll) : KiUserExceptionDispatcher
19) 00007FFB49FB99D6 (LandscapeCanvas.Editor) : UninitializeDynamicModule
18) 00007FFB49FB90F6 (LandscapeCanvas.Editor) : UninitializeDynamicModule
17) 00007FFB49FE437D (LandscapeCanvas.Editor) : UninitializeDynamicModule
16) 00007FFB49FEB121 (LandscapeCanvas.Editor) : UninitializeDynamicModule
15) 00007FFB4AE4679C (GraphModel.Editor) : UninitializeDynamicModule
14) 00007FFB4AE7BEA7 (GraphModel.Editor) : UninitializeDynamicModule
13) 00007FFB5BCB57A4 (GraphCanvas.Editor) : (function-name not available)
12) 00007FFB5BCE5F0F (GraphCanvas.Editor) : (function-name not available)
11) 00007FFB4AE48A8F (GraphModel.Editor) : UninitializeDynamicModule
10) 00007FFB4AE6BFB5 (GraphModel.Editor) : UninitializeDynamicModule
 9) 00007FFB4AE67E0F (GraphModel.Editor) : UninitializeDynamicModule
 8) 00007FFB49F7426B (LandscapeCanvas.Editor) : UninitializeDynamicModule
 7) 00007FFB49FD72BC (LandscapeCanvas.Editor) : UninitializeDynamicModule
 6) 00007FFB5BCFD1D3 (GraphCanvas.Editor) : (function-name not available)
 5) 00007FFB5BCB46E4 (GraphCanvas.Editor) : (function-name not available)
 4) 00007FFB5BD12F51 (GraphCanvas.Editor) : (function-name not available)
 3) 00007FFB84D49677 (Qt5Widgets) : QGraphicsScene::event
 2) 00007FFB84A849B0 (Qt5Widgets) : QApplicationPrivate::notify_helper
 1) 00007FFB84A83A33 (Qt5Widgets) : QApplication::notify
```

**Video**

https://user-images.githubusercontent.com/86951967/133626430-defe9261-4e07-4d29-9c31-153b5e765b4c.mp4

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version: 10
- CPU Intel Core i7-8700K
- GPU: NVIDIA GeForce RTX 2080 Ti
- Memory: 16GB

**Additional Context**
[error.zip](https://github.com/o3de/o3de/files/7178372/error.zip)
[error.log](https://github.com/o3de/o3de/files/7178373/error.log)
[Editor.log](https://github.com/o3de/o3de/files/7178377/Editor.log)

**Comment by alexmontAmazon** [comment]
@cgalvan would you take a look at this please?

**Comment by LB-JakubSkorupka** [comment]
Verified as fixed on Development (2c55ad5)
